### PR TITLE
Store arbitrary number of commitments on channelState

### PR DESCRIPTION
**Pull request by andrewgordstewart**
This feature is required to implement virtual channels.

Some commits unrelated to this feature are included for my convenience, which make a little progress towards the AdvanceChannel protocol.

### CODE REVIEW

File: packages/wallet/src/redux/types.ts [hunk 1]
```diff
@@ -1,4 +1,10 @@
-export const enum PlayerIndex {
+export const enum TwoPartyPlayerIndex {
   A = 0,
   B = 1,
 }
+
+export const enum ThreePartyPlayerIndex {
```
Comment (kerzhner) [inline packages/wallet/src/redux/types.ts]: Do we foresee any cases with 4 parties coming up? Should we start moving away from two and three party cases toward n-party structures? Or is the added complexity not worth it?

File: packages/wallet/src/redux/protocols/reducer-helpers.ts [hunk 2]
```diff
@@ -45,18 +45,13 @@ export const filterOutSignatureMessages = (sideEffects?: SideEffects): SideEffec
 
 export function sendFundingComplete(sharedData: SharedData, appChannelId: string) {
   const channelState = selectors.getOpenedChannelState(sharedData, appChannelId);
-  const lastCommitment = channelState.lastCommitment.commitment;
-  if (
-    lastCommitment.commitmentType !== CommitmentType.PostFundSetup ||
-    lastCommitment.turnNum !== 3
-  ) {
+  const c = getLastCommitment(channelState);
```
Comment (kerzhner) [inline packages/wallet/src/redux/protocols/reducer-helpers.ts]: Not a fan of one letter variable names.

File: packages/wallet/src/redux/protocols/dispute/challenger/__tests__/scenarios.ts [hunk 3]
```diff
@@ -32,14 +29,18 @@ const {
   signedCommitment21,
 } = channelScenarios;
 
-const partiallyOpen = partiallyOpenChannelFromCommitment(signedCommitment0, address, privateKey);
+const partiallyOpen = channelFromCommitments([signedCommitment0], address, privateKey);
 const theirTurn = channelFromCommitments(
-  signedCommitment19,
-  signedCommitment20,
+  [signedCommitment19, signedCommitment20],
+
```
Comment (kerzhner) [inline packages/wallet/src/redux/protocols/dispute/challenger/__tests__/scenarios.ts]:
```suggestion
```

File: packages/wallet/src/redux/types.ts (hunk 1 above)
Comment (andrewgordstewart) [reply packages/wallet/src/redux/types.ts]: I don't know if we'll ever get rid of all of the reliance on two parties. If we do, then we should get rid of the `*PlayerIndex` types, for sure.

**Pull request merged**